package.json

```diff
@@ -40,15 +40,15 @@
     ]
   },
   "scripts": {
-    "preinstall": "node automation/check-npm-version.js",
     "postinstall": "patch-package",
-    "prebuild": "rimraf build/ build-bin/",
+    "prebuild": "npm run npmcheck && rimraf build/ build-bin/",
```

Comment:
I am of two minds about this PR and I don't know if it should be merged:
* In favour: it allows CLI end users (as opposed to CLI developers) to install the CLI with npm v6.4.1 that ships with Node 8. It will damage their `npm-shrinkwrap.json` file but it's OK, because the damaged file won't be reused.
* Against: it allows a CLI contributor to submit a PR with a broken `npm-shrinkwrap.json` file that could end up being merged undetected, then breaking the CLI for all users. This is possible because merely running `npm install` damages the file, and the CLI contributor does not strictly have to run `npm run build` or `npm test` in order to submit a PR. **For example, a PR that merely updates the markdown documentation files (README, INSTALL...).** The CI will run `npm install` and `npm test` with a good npm version, but this neither detects the issue nor undamages the `npm-shrinkwrap.json` file.

Reply:
I've decided to merge this PR sooner than later, to avoid users' CIs out there breaking unnecessarily with Node 8. We can still revisit this decision and readopt preinstall. I am also considering that until yesterday, we had no checks at all, and having a husky pre-commit check is an improvement over that.

I've tested all installation methods in the draft release for v11.16.4.